From 66d6e56e67db880b40b8c4fe3118698d8244b657 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Fri, 26 Jun 2015 23:07:19 +0800
Subject: [PATCH] MLK-11178-02 thermal: imx: disbale hardware reset funtion on
 imx7

As per to the design team, use the software poweroff when critical
trip temp reached. No need to set the panic alam register bits
when using the software poweroff as the critical trip action.

Signed-off-by: Bai Ping <b51503@freescale.com>
---
 drivers/thermal/imx_thermal.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/thermal/imx_thermal.c b/drivers/thermal/imx_thermal.c
index b4fafad..56ca4ab 100644
--- a/drivers/thermal/imx_thermal.c
+++ b/drivers/thermal/imx_thermal.c
@@ -393,7 +393,7 @@ static int imx_set_trip_temp(struct thermal_zone_device *tz, int trip,
 
 	if (trip == IMX_TRIP_CRITICAL) {
 		data->temp_critical = temp;
-		if (data->socdata->version != TEMPMON_V1)
+		if (data->socdata->version == TEMPMON_V2)
 			imx_set_panic_temp(data, temp);
 	}
 
@@ -756,8 +756,7 @@ static int imx_thermal_probe(struct platform_device *pdev)
 
 	/* Enable measurements at ~ 10 Hz */
 	measure_freq = DIV_ROUND_UP(32768, 10); /* 10 Hz */
-	if (data->socdata->version != TEMPMON_V3)
-		regmap_field_write(data->measure_freq, measure_freq);
+	regmap_field_write(data->measure_freq, measure_freq);
 	imx_set_alarm_temp(data, data->temp_passive);
 
 	if (data->socdata->version == TEMPMON_V2)
-- 
1.8.0

